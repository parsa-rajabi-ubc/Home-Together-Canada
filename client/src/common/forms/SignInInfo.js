/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.05
 *
 * @Description: Sign in input Form Component
 *
 */

import React from "react";
import PropTypes from "prop-types";
import Tooltip from "./Tooltip";

const INFO_TEXT = {
    USERNAME: "Your username should be unique and meaningful. Avoid using numbers in your username!",
    PASSWORD: "Your password must be at least 8 characters long and contain a mix of numeric digits, upper and lower case letters!",
    CONFIRM_PASSWORD: "This password must match the password you entered above!"
};


function SignInInfo(props) {
    const {onChangeUsername, onChangePassword, onChangePasswordCheck} = props;
    return (
        <div>
            <label>
                <span className={"label"}>Username</span>
                <Tooltip text={INFO_TEXT["USERNAME"]} toolTipID="username"/>
                <input className={"input"} type="text" placeholder="" onChange={onChangeUsername}/>
            </label>
            <label>
                <span className={"label"}>Password</span>
                <Tooltip text={INFO_TEXT["PASSWORD"]} toolTipID="password"/>
                <input className={"input"} type="password" placeholder="" onChange={onChangePassword}/>
            </label>
            <label>
                <span className={"label"}>Confirm Password</span>
                <Tooltip text={INFO_TEXT["CONFIRM_PASSWORD"]} toolTipID="confirmPassword"/>
                <input className={"input"} type="password" placeholder="" onChange={onChangePasswordCheck}/>
            </label>
        </div>
    );
}

SignInInfo.propTypes = {
    onChangeUsername: PropTypes.func.isRequired,
    onChangePassword: PropTypes.func.isRequired,
    onChangePasswordCheck: PropTypes.func.isRequired
};

export default SignInInfo;