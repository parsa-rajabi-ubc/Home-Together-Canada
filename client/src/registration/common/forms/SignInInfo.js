import React from "react";

function SignInInfo(){
    return(
        <div>
            <label>
                Username:
                <input type="text" placeholder="" />
            </label>
            <label>
                Password:
                <input type="text" placeholder="" />
            </label>
            <label>
                Confirm Password:
                <input type="text" placeholder="" />
            </label>
        </div>
    );
}

export default SignInInfo