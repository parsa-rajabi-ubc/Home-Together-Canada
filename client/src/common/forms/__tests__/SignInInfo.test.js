/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.13
 *
 * @Description: Sign in info Component Snapshot test
 *
 */
import React from "react";
import SignInInfo from "../SignInInfo";
import renderer from 'react-test-renderer';
jest.mock("react-tooltip/node_modules/uuid", () => ({ v4: () => "00000000-0000-0000-0000-000000000000" }));

describe('SignInInfo', () => {
        describe('Snapshot test', () => {
                it("should match stored snapshot", () => {
                        // given
                        const onChangeUsername = jest.fn();
                        const onChangePassword = jest.fn();
                        const onChangePasswordCheck = jest.fn();

                        //when
                        const component = renderer.create(<SignInInfo onChangeUsername={onChangeUsername} onChangePassword={onChangePassword} onChangePasswordCheck={onChangePasswordCheck}/>);

                        // then
                        expect(component).toMatchSnapshot();
                });
        })
})