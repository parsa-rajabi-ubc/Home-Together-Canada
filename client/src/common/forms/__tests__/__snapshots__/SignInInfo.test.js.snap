// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInInfo Snapshot test should match stored snapshot 1`] = `
<div>
  <label>
    <label
      className="label"
    >
      Username
       
      <svg
        className="inline align-top text-red-600"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 6H13V10.079L16.3413 7.73938L17.4885 9.37768L13.7434 12L17.4885 14.6223L16.3413 16.2606L13 13.921V18H11V13.921L7.65864 16.2606L6.51148 14.6223L10.2565 12L6.51147 9.37769L7.65863 7.73938L11 10.079V6Z"
          fill="currentColor"
        />
      </svg>
    </label>
    <span>
      <a
        data-for="username"
        data-tip={true}
      >
        <svg
          className="inline ml-1 mb-1 text-blue-500 hover:text-blue-400"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M2 0a2 2 0 00-2 2v12a2 2 0 002 2h12a2 2 0 002-2V2a2 2 0 00-2-2H2zm4.57 6.033H5.25C5.22 4.147 6.68 3.5 8.006 3.5c1.397 0 2.673.73 2.673 2.24 0 1.08-.635 1.594-1.244 2.057-.737.559-1.01.768-1.01 1.486v.355H7.117l-.007-.463c-.038-.927.495-1.498 1.168-1.987.59-.444.965-.736.965-1.371 0-.825-.628-1.168-1.314-1.168-.901 0-1.358.603-1.358 1.384zm1.251 6.443c-.584 0-1.009-.394-1.009-.927 0-.552.425-.94 1.01-.94.609 0 1.028.388 1.028.94 0 .533-.42.927-1.029.927z"
            fillRule="evenodd"
          />
        </svg>
      </a>
      <div
        className="__react_component_tooltip t00000000-0000-0000-0000-000000000000 place-top type-dark"
        data-id="tooltip"
        id="username"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": "
  	.t00000000-0000-0000-0000-000000000000 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t00000000-0000-0000-0000-000000000000.place-top {
        margin-top: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::before {
        border-top: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-bottom {
        margin-top: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-left {
        margin-left: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::before {
        border-left: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-right {
        margin-left: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::before {
        border-right: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
            }
          }
        />
        Your username should be unique and meaningful.
      </div>
    </span>
    <input
      onChange={[MockFunction]}
      placeholder=""
      type="text"
    />
  </label>
  <label>
    <label
      className="label"
    >
      Password
       
      <svg
        className="inline align-top text-red-600"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 6H13V10.079L16.3413 7.73938L17.4885 9.37768L13.7434 12L17.4885 14.6223L16.3413 16.2606L13 13.921V18H11V13.921L7.65864 16.2606L6.51148 14.6223L10.2565 12L6.51147 9.37769L7.65863 7.73938L11 10.079V6Z"
          fill="currentColor"
        />
      </svg>
    </label>
    <span>
      <a
        data-for="password"
        data-tip={true}
      >
        <svg
          className="inline ml-1 mb-1 text-blue-500 hover:text-blue-400"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clipRule="evenodd"
            d="M2 0a2 2 0 00-2 2v12a2 2 0 002 2h12a2 2 0 002-2V2a2 2 0 00-2-2H2zm4.57 6.033H5.25C5.22 4.147 6.68 3.5 8.006 3.5c1.397 0 2.673.73 2.673 2.24 0 1.08-.635 1.594-1.244 2.057-.737.559-1.01.768-1.01 1.486v.355H7.117l-.007-.463c-.038-.927.495-1.498 1.168-1.987.59-.444.965-.736.965-1.371 0-.825-.628-1.168-1.314-1.168-.901 0-1.358.603-1.358 1.384zm1.251 6.443c-.584 0-1.009-.394-1.009-.927 0-.552.425-.94 1.01-.94.609 0 1.028.388 1.028.94 0 .533-.42.927-1.029.927z"
            fillRule="evenodd"
          />
        </svg>
      </a>
      <div
        className="__react_component_tooltip t00000000-0000-0000-0000-000000000000 place-top type-dark"
        data-id="tooltip"
        id="password"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": "
  	.t00000000-0000-0000-0000-000000000000 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
  	}

  	.t00000000-0000-0000-0000-000000000000.place-top {
        margin-top: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::before {
        border-top: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-top::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        bottom: -6px;
        left: 50%;
        margin-left: -8px;
        border-top-color: #222;
        border-top-style: solid;
        border-top-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-bottom {
        margin-top: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::before {
        border-bottom: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-bottom::after {
        border-left: 8px solid transparent;
        border-right: 8px solid transparent;
        top: -6px;
        left: 50%;
        margin-left: -8px;
        border-bottom-color: #222;
        border-bottom-style: solid;
        border-bottom-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-left {
        margin-left: -10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::before {
        border-left: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-left::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        right: -6px;
        top: 50%;
        margin-top: -4px;
        border-left-color: #222;
        border-left-style: solid;
        border-left-width: 6px;
    }

    .t00000000-0000-0000-0000-000000000000.place-right {
        margin-left: 10px;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::before {
        border-right: 8px solid transparent;
    }
    .t00000000-0000-0000-0000-000000000000.place-right::after {
        border-top: 5px solid transparent;
        border-bottom: 5px solid transparent;
        left: -6px;
        top: 50%;
        margin-top: -4px;
        border-right-color: #222;
        border-right-style: solid;
        border-right-width: 6px;
    }
  ",
            }
          }
        />
        Your password must be at least 8 characters long and contain a mix of numeric digits, upper and lower case letters!
      </div>
    </span>
    <input
      onChange={[MockFunction]}
      placeholder=""
      type="password"
    />
  </label>
  <label>
    <label
      className="label"
    >
      Confirm Password
       
      <svg
        className="inline align-top text-red-600"
        fill="none"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 6H13V10.079L16.3413 7.73938L17.4885 9.37768L13.7434 12L17.4885 14.6223L16.3413 16.2606L13 13.921V18H11V13.921L7.65864 16.2606L6.51148 14.6223L10.2565 12L6.51147 9.37769L7.65863 7.73938L11 10.079V6Z"
          fill="currentColor"
        />
      </svg>
    </label>
    <input
      onChange={[MockFunction]}
      placeholder=""
      type="password"
    />
  </label>
</div>
`;
